open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/JWTUtils.java

```diff
@@ -133,7 +135,19 @@ public static boolean validateJWTSignature(String jwtString, String jwksUri, Str
         jwtProcessor.setJWSKeySelector(keySelector);
         // Process the token, set optional context parameters.
         SimpleSecurityContext securityContext = new SimpleSecurityContext();
-        jwtProcessor.process((SignedJWT) jwt, securityContext);
+        JWTClaimsSet claims = jwtProcessor.process((SignedJWT) jwt, securityContext);
+
+        // Validate iat with configured timestamp skew if validation is enabled and iat is sent in the request.
+        boolean validateIat = OpenBankingConfigParser.getInstance().isDCRIATValidationEnabled();
+        Date iat = claims.getIssueTime();
+        if (validateIat && iat != null) {
+            long timestampSkew = java.util.concurrent.TimeUnit.SECONDS.toMillis(
+                    OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds());
+            long now = System.currentTimeMillis();
+            if (iat.getTime() > now + Math.max(0, timestampSkew)) {
```

Comment:
Do we need to handle the situation where this is just a number and not a valid timestamp?